### DISCUSSION TITLE
Bitwarden Send : error

### DISCUSSION THREAD

**Discussion by doreks** [dani-garcia/vaultwarden#1498] [Q&A]
### Subject of the issue

using the new bitwarden send feature I am getting this error:

```
_An error has occurred
<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8"> <title>404 Not Found</title> </head> <body align="center"> <div role="main" align="center"> <h1>404: Not Found</h1> <p>The requested resource could not be found.</p> <hr /> </div> <div role="contentinfo" align="center"> <small>Rocket</small> </div> </body> </html>_
```

### Your environment (Generated via diagnostics page)
* Bitwarden_rs version: v1.19.0
* Web-vault version: v2.18.1
* Running within Docker: true
* Internet access: true
* Uses a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.*********.***:****",
  "domain_origin": "*****://*********.*********.***:****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Bitwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "log_file": "/data/bitwarden.log",
  "log_level": "warn",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "show_password_hint": true,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": true,
  "smtp_from": "********@*****.***",
  "smtp_from_name": "Bitwarden",
  "smtp_host": "****.*****.***",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "********@*****.***",
  "templates_folder": "data/templates",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

* bitwarden_rs version:

* Install method:
docker installation:
docker run -d --name='bitwardenrs' --net='private-net' --ip='172.22.0.22' -e TZ="Europe/Paris" -e HOST_OS="Unraid" -e 'SERVER_ADMIN_EMAIL'='email@email.com' -e 'SIGNUPS_ALLOWED'='false' -e 'INVITATIONS_ALLOWED'='false' -e 'WEBSOCKET_ENABLED'='true' -e 'ADMIN_TOKEN'='*********' -p '8086:80/tcp' -v '/mnt/user/appdata/bitwarden':'/data':'rw' -e LOG_LEVEL=warn -e EXTENDED_LOGGING=true -e LOG_FILE=/data/bitwarden.log -e WEBSOCKET_ENABLED=true -p 3012:3012 'bitwardenrs/server'

* Clients used: 
desktop

**Comment by darrenjl0** [comment]
Send functionality was only recently merged (implemented) in the testing branch not "bitwardenrs/**server**" which is the latest stable release. Pretty sure the dev's said you can run the testing branch if you really want send functionality now. You can still downgrade later if you want as there is no breaking database changes but I would backup 1st anyway.

**Comment by BlackDex** [comment]
use `bitwardenrs/server:testing` if you want to have this function.

**Reply by blaine07** [reply]
I know things take due process... but pardon my idiocracy here. How long would something like “send implementation” usually spending in the “testing” department out of curiousity?

**Reply by BlackDex** [reply]
We do not have any timelines, milestones or something like that. So it's more a feeling i think, depending a bit on the features added, PR's still open which maybe could be merged or not.

So there is actually nothing to say definitely about it.

**Reply by blaine07** [reply]
Thanks mate 😀
